components/SearchBar/index.js

```diff
@@ -11,12 +25,25 @@ function SearchBar({ query, setQuery, handleSearchInput, searchResults }) {
 
     setQuery(searchQuery.searchInput);
     handleSearchInput(searchQuery.searchInput);
+    setHasEntries(true);
+
+    // Search Dictionary if no searchResults in list
+    if (searchResults.length === 0 && query.length > 0) {
+      setDictionaryQuery(searchQuery.searchInput);
+      setHasEntries(false);
+    }
```

Comment:
`searchResults` is a state that is **not immediately updated** after the function `handleSearchInput` has completed. Setting a new state is asynchronous. If you check `searchResults` here, it's still the old state you're checking.

Reply:
That would do the trick, yes. The `useEffect` should be declared in the same file as where the `searchResults` lives.

Try it, I hope it won't break anything else in the process 😀 
If it does break something, you may as well leave it out entirely. The above code snipped doesn't do anything either if I'm not mistaken.

Reply:
All right, let's leave it then!